blockchain/src/block_production/mod.rs

```diff
@@ -13,9 +13,43 @@ use nimiq_transaction::{
     historic_transaction::HistoricTransaction, inherent::Inherent, Transaction,
 };
 use rand::{CryptoRng, Rng, RngCore};
+use thiserror::Error;
 
 use crate::{interface::HistoryInterface, Blockchain};
 
+#[derive(Debug, Error)]
+pub enum BlockProducerError {
+    #[error("Failed to commit: error={0}, account={1:?}, transactions={2:?}, inherents={3:?}")]
+    AccountsError(AccountsError, Account, Vec<Transaction>, Vec<Inherent>),
+    #[error("Failed to add to history")]
+    HistoryError,
+    #[error("Accounts are incomplete")]
+    AccountsIncomplete,
+}
+
+impl BlockProducerError {
+    fn accounts_error(
+        blockchain: &Blockchain,
+        error: AccountsError,
```

Comment:
```suggestion
    fn from_accounts_error(
        error: AccountsError,
        blockchain: &Blockchain,
```
I like this better, it is not a deal breaker though.